harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -636,3 +666,77 @@ def test_checkpoint_loading(ckpt: str, istrial: bool):
         assert isinstance(trial, pytorch.PyTorchTrial), type(trial)
     else:
         assert isinstance(trial, torch.nn.Module), type(trial)
+
+
+def make_amp_workloads(
+    trial_class,
+    assert_output_float16=False,
+) -> workload.Stream:
+    trainer = utils.TrainAndValidate()
+    yield from trainer.send(steps=20, validation_freq=1, scheduling_unit=1)
+    training_metrics, _ = trainer.result()
+
+    for metrics, next_metrics in zip(training_metrics[:-1], training_metrics[1:]):
+        scale = next_metrics["scale_before"]
+        if "scale" in metrics:
+            assert metrics["scale"] == scale
+        else:
+            metrics["scale"] = scale
+        assert "scale" in metrics
+    loss_prev = None
+    GROWTH_INTERVAL = trial_class._growth_interval
+    growth_countdown = GROWTH_INTERVAL
+    # Only attempt assertions up to and including the penultimate batch, because
+    #  we may not have the updated scale from the final batch.
+    for idx, metrics in enumerate(training_metrics[:-1]):
+        assert metrics["loss"].dtype is np.dtype("float32")
+        if assert_output_float16:
+            assert metrics["output"].dtype is np.dtype("float16"), (
+                f"output has dtype {metrics['output'].dtype} but should be float16"
+            )
+        else:
+            # Automatic usages of Apex or native AMP cast the output back to float32
+            # For the latter case, see the hook end_f16
+            #   defined in PyTorchTrialContext.autocast_forward_pass
+            assert metrics["output"].dtype is np.dtype("float32"), (
+                f"output has dtype {metrics['output'].dtype} but should be float32"
+            )
+        loss = metrics["loss"].item()
+        scale_before = metrics["scale_before"]
+        scaled_loss = loss * scale_before
+        if "scaled_loss" in metrics:
+            assert scaled_loss == metrics["scaled_loss"]
```

Comment:
this assertion strikes me as not testing any code path in PyTorchTrial, since the only time `scaled_loss` is set is in the manual case.

Reply:
You're right. I removed this and other related code in 7e6ac53.